app/web/src/newhotness/StatusPanel.vue

```diff
@@ -0,0 +1,84 @@
+<template>
+  <div class="mt-xs ml-xs relative">
+    <StatusPanelIcon :status="status" @faded="() => (status = undefined)" />
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { computed, ref, reactive, watch, onMounted, onUnmounted } from "vue";
+import { useQuery } from "@tanstack/vue-query";
+import StatusPanelIcon from "@/newhotness/StatusPanelIcon.vue";
+import { useMakeKey, bifrost, useMakeArgs } from "@/store/realtime/heimdall";
+import {
+  DependentValueComponentList,
+  EntityKind,
+} from "@/workers/types/entity_kind_types";
+
+const bucket = reactive<Record<string, number>>({});
+const trigger = ref<boolean>(false);
+const status = ref<"syncing" | "synced" | undefined>(undefined);
+
+const timeoutMs = 30000;
+const tickMs = 1000;
+
+const key = useMakeKey();
+const args = useMakeArgs();
+const dependentValueComponentListQuery =
+  useQuery<DependentValueComponentList | null>({
+    queryKey: key(EntityKind.DependentValueComponentList),
+    queryFn: async () =>
+      await bifrost<DependentValueComponentList>(
+        args(EntityKind.DependentValueComponentList),
+      ),
+  });
+const componentsInFlight = computed(
+  () => dependentValueComponentListQuery.data.value?.componentIds ?? [],
+);
+
+watch(
+  () => componentsInFlight.value,
+  (newComponentsInFlight) => {
+    if (newComponentsInFlight.length > 0 && !bucket.componentsInFlight) {
+      bucket.componentsInFlight = Date.now();
+    } else if (newComponentsInFlight.length < 1 && bucket.componentsInFlight) {
+      delete bucket.componentsInFlight;
+    }
+  },
+);
```

Comment:
This is our first bucket item: components or flights. I'd love to use a set here with just timers, but knowing _**what**_ is being tracked will be important soon.